phpBB/phpbb/ucp/controller/reset_password.php

```diff
@@ -0,0 +1,436 @@
+		$this->db = $db;
+		$this->dispatcher = $dispatcher;
+		$this->helper = $helper;
+		$this->language = $language;
+		$this->log = $log;
+		$this->passwords_manager = $passwords_manager;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+		$this->tables = $tables;
+		$this->root_path = $root_path;
+		$this->php_ext = $php_ext;
+	}
+
+	/**
+	 * Init controller
+	 */
+	protected function init_controller()
+	{
+		$this->language->add_lang('ucp');
+
+		if (!$this->config['allow_password_reset'])
+		{
+			trigger_error($this->language->lang('UCP_PASSWORD_RESET_DISABLED', '<a href="mailto:' . htmlspecialchars($this->config['board_contact']) . '">', '</a>'));
+		}
+	}
+
+	/**
+	 * Remove reset token for specified user
+	 *
+	 * @param int $user_id User ID
+	 */
+	protected function remove_reset_token(int $user_id)
+	{
+		$sql_ary = [
+			'reset_token'				=> '',
+			'reset_token_expiration'	=> 0,
+		];
+
+		$sql = 'UPDATE ' . $this->tables['users'] . '
+					SET ' . $this->db->sql_build_array('UPDATE', $sql_ary) . '
+					WHERE user_id = ' . $user_id;
+		$this->db->sql_query($sql);
+	}
+
+	/**
+	 * Handle password reset request
+	 *
+	 * @return Response
+	 */
+	public function request()
+	{
+		$this->init_controller();
+
+		$submit		= $this->request->is_set_post('submit');
+		$username	= $this->request->variable('username', '', true);
+		$email		= strtolower($this->request->variable('email', ''));
+
+		add_form_key('ucp_reset_password');
+
+		if ($submit)
+		{
+			if (!check_form_key('ucp_reset_password'))
+			{
+				return $this->helper->message('FORM_INVALID');
+			}
+
+			if (empty($email))
+			{
+				return $this->helper->message('NO_EMAIL_USER');
+			}
+
+			$sql_array = [
+				'SELECT'	=> 'user_id, username, user_permissions, user_email, user_jabber, user_notify_type, user_type,'
+								. ' user_lang, user_inactive_reason, reset_token, reset_token_expiration',
+				'FROM'		=> [$this->tables['users'] => 'u'],
+				'WHERE'		=> "user_email_hash = '" . $this->db->sql_escape(phpbb_email_hash($email)) . "'" .
+					(!empty($username) ? " AND username_clean = '" . $this->db->sql_escape(utf8_clean_string($username)) . "'" : ''),
+			];
+
+			/**
+			 * Change SQL query for fetching user data
+			 *
+			 * @event core.ucp_remind_modify_select_sql
+			 * @var	string	email		User's email from the form
+			 * @var	string	username	User's username from the form
+			 * @var	array	sql_array	Fully assembled SQL query with keys SELECT, FROM, WHERE
+			 * @since 3.1.11-RC1
+			 * @changed 3.3.0-b1 Moved to reset password controller
+			 */
+			$vars = [
+				'email',
+				'username',
+				'sql_array',
+			];
+			extract($this->dispatcher->trigger_event('core.ucp_remind_modify_select_sql', compact($vars)));
+
+			$sql = $this->db->sql_build_query('SELECT', $sql_array);
+			$result = $this->db->sql_query_limit($sql, 2); // don't waste resources on more rows than we need
+			$rowset = $this->db->sql_fetchrowset($result);
+
```

Comment:
I would just free the result here, you already have the entire `$rowset` anyway.